### PR TITLE
Fixes task deletion and adds completion toggle functionality

### PR DESCRIPTION
**Pull request by Tepanyega**
Fixed Task Deletion:
Updated the task deletion endpoint to properly remove tasks from the array.
Tasks are now correctly deleted when the delete icon is clicked.

Added Completion Toggle:
Implemented a feature to mark tasks as completed. When a user clicks the Complete button, the task text is crossed out (strikethrough) to indicate completion.

Why this change is needed:
The task deletion was not functioning correctly in the previous version due to a bug in the route handling.
The addition of a completion toggle enhances the user experience, allowing users to track completed tasks visually.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## Getting Started
+
+### 1. Clone the repository
+
+Clone the repository to your local machine:
+
+```bash
+git clone https://github.com/yourusername/to-do-list.git
+```
```
Comment (bokangkhumalo) [inline README.md:35]: Hi @Tepanyega 

Thank you for your contributions! I wanted to point out that your clone link says yourusername, please fix this omission

**Review by bokangkhumalo — CHANGES_REQUESTED** [review]
Hi @Tepanyega,

Great work on implementing the core functionality! I have a few suggestions that could make the code easier to understand and maintain:

Add Comments for Key Sections: The absence of comments makes some parts of the code difficult to follow, particularly around complex logic and data processing steps. Brief comments summarizing each key section would enhance readability.

Clarify Function and Variable Names: Some functions and variables (e.g., doTask()) could be renamed to better reflect their specific purposes. This can improve understanding without needing additional comments.

Add Documentation or README: A short README or module-level docstring explaining the project structure and the role of each module would be very helpful for future contributors and reviewers.

File: README.md (hunk 1 above)
Comment (Tepanyega) [reply README.md:35]: Hi @bokangkhumalo,

Sorted, Thank you.

File: public/css/style.css
Comment (alphineandre) [inline public/css/style.css:1]: I like everything so far but i would say improve the CSS, at this moment in time it looks a bit like a default webpage give it a bit more color and try separating the current tasks and the finished tasks.

**Review by alphineandre — CHANGES_REQUESTED** [review]
I think if u can make those changes it will be complete

File: README.md (hunk 1 above)
Comment (bokangkhumalo) [reply README.md:36]: HI @Tepanyega,

Glad to see you implement the changes! happy coding

**Review by bokangkhumalo — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bokangkhumalo) [reply README.md:36]: g

**Review by KhanyiCodes — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ Before running the app, ensure you have the following installed:
 Clone the repository to your local machine:
 
 ```bash
-git clone https://github.com/yourusername/to-do-list.git
+git clone https://github.com/Tepanyega/to_doList.git
```
Comment (bokangkhumalo) [inline README.md]: Hi @Tepanyega, Glad to see your changes. Happy coding! ;)

**Review by bokangkhumalo — APPROVED** [review]

File: public/css/style.css
Comment (Tepanyega) [reply public/css/style.css:1]: Thank you, 
I made some changes on the design, you can check it out.

Comment (KhanyiCodes) [reply public/css/style.css:1]: Thank you for your feedback! I'm really glad to hear you're liking the app so far. I totally agree that the CSS could use some improvement to make it feel more polished and visually appealing. To enhance the design, I’ve decided to incorporate neon colors into the layout. These bright, vibrant hues will add more energy and create a more dynamic, engaging interface.

For separating current tasks from finished tasks, I think your suggestion is spot-on. I'll work on creating a clearer visual distinction between them. Here’s what I’m thinking:

Neon-colored highlights or labels to differentiate tasks based on their status.
A subtle neon border or background shade for completed tasks to make them stand out.
Interactive elements like buttons or status indicators that use neon tones to create a modern, eye-catching effect.
I’ll be sure to balance the neon elements so it doesn’t overwhelm the user, but instead creates a sleek, vibrant, and functional experience. Once I’ve made these adjustments, I'll share an updated version for your feedback.

Thanks again for the great suggestions! Let me know if there’s anything else you’d like me to focus on, and I’ll keep you posted on the progress.

New CSS:
image

**Review by alphineandre — APPROVED** [review]
Great work its looks great now

**Review by KhanyiCodes — APPROVED** [review]

**Review by KhanyiCodes — APPROVED** [review]

**Review by PhodzoW0l7 — APPROVED** [review]

**Review by JKRakuba — APPROVED** [review]

**Pull request merged**